srcjs/shinyapp.js

```diff
@@ -331,18 +331,22 @@ var ShinyApp = function() {
   };
 
   this.receiveOutput = function(name, value) {
-    if (this.$values[name] === value)
-      return undefined;
-
-    this.$values[name] = value;
-    delete this.$errors[name];
-
     var binding = this.$bindings[name];
     var evt = jQuery.Event('shiny:value');
     evt.name = name;
     evt.value = value;
     evt.binding = binding;
+
+    if (this.$values[name] === value) {
+      $(binding ? binding.el : document).trigger(evt);
+      return undefined;
+    }
+
```

Comment:
The order is incorrect; the event shouldn't be triggered until after `this.$values[name] = value;` and `delete this.$errors[name];`. If for no other reason, errors in the trigger shouldn't stop those side effects.

Reply:
There are two cases, as in the version without the PR: a) client receives the same value as before, b) client receives a new value.

In case a) there is no need to assign the value (this should be more efficient as the DOM doesn't need to be updated) and there should be no errors to delete. This is the same behaviour as before, the only change here is that the event is now triggered (the event parameters need to be set up before).